physical/raft/raft.go

```diff
@@ -1232,6 +1232,63 @@ func (b *RaftBackend) RemovePeer(ctx context.Context, peerID string) error {
 	return b.autopilot.RemoveServer(raft.ServerID(peerID))
 }
 
+// PromotePeer promotes a permanent non-voter to voter
+func (b *RaftBackend) PromotePeer(ctx context.Context, peerID string) error {
+	b.l.RLock()
+	defer b.l.RUnlock()
+
+	if err := ctx.Err(); err != nil {
+		return err
+	}
+
+	if b.disableAutopilot {
+		if b.raft == nil {
+			return errors.New("raft storage is not initialized")
+		}
+		return errors.New("raft storage autopilot must be enabled to promote a peer")
```

Comment:
I thought we also supported manual joining of servers as non-voter nodes, outside of auto-pilot?

In the CLI for `bao operator raft join`:

```go
    f.BoolVar(&BoolVar{
        Name:    "non-voter",
        Target:  &c.flagNonVoter,
        Default: false,
        Usage:   "Join the Raft cluster as non-voter.",
    })
```

Can we not promote/demote these as well?

Reply:
Hmm I think we might be able to do that, but without autopilot we can't really guarantee the promoted/demoted node stays in it's given role. I wonder if there's a valid use-case of running OpenBao without Raft autopilot at all :thinking:

Reply:
How is auto-pilot state persisted? I thought we had storage level state that stored voter/non-voter status that we needed to update?

Reply:
We do, but this is managed by the Raft Autopilot Delegate here:
https://github.com/openbao/openbao/blob/main/physical/raft/raft_autopilot_delegate.go#L35
And persisted to the Raft store here:
https://github.com/openbao/openbao/blob/main/physical/raft/raft_autopilot_delegate.go#L309-L324